### PR TITLE
1223 Add get and link QID endpoints

### PR DESCRIPTION
**Pull request by AdamHawtin**
# Motivation and Context
We need endpoints which can fetch the details and link QIDs to cases.

# What has changed
* Add get QID endpoint
* Add link QID endpoint

# How to test?
Build and run with branch AT's

# Links
https://trello.com/c/6QdthBHC/1223-r-ops-ui-link-a-qid-to-a-case-13

### CODE REVIEW

File: src/main/java/uk/gov/ons/census/caseapisvc/model/dto/QidLink.java [hunk 1]
```diff
@@ -0,0 +1,12 @@
+package uk.gov.ons.census.caseapisvc.model.dto;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import java.util.UUID;
+import lombok.Data;
+
+@Data
+@JsonInclude(JsonInclude.Include.NON_NULL)
+public class QidLink {
```
Comment (NickGrantX) [inline src/main/java/uk/gov/ons/census/caseapisvc/model/dto/QidLink.java:9]: Needs to include the desired transaction ID, which will also be audit logged by the ROps UI

File: src/main/java/uk/gov/ons/census/caseapisvc/service/UacQidService.java [hunk 2]
```diff
@@ -101,4 +132,24 @@ private static boolean isHouseholdCaseType(String caseType) {
   private static boolean isCeCaseType(String caseType) {
     return caseType.equals(CASE_TYPE_CE);
   }
+
+  public void buildAndSendQuestionnaireLinkedEvent(UacQidLink uacQidLink, Case caseToLink) {
+    UacDTO uacDTO = new UacDTO();
+    uacDTO.setCaseId(caseToLink.getCaseId());
+    uacDTO.setQuestionnaireId(uacQidLink.getQid());
+
+    EventDTO eventDTO = new EventDTO();
```
Comment (NickGrantX) [inline src/main/java/uk/gov/ons/census/caseapisvc/service/UacQidService.java:138]: I'm tempted to say that the channel should be "ROPS_UI" to differentiate this event from stuff that's been done by the contact centre

**Review by NickGrantX — CHANGES_REQUESTED** [review]
See comments

File: src/test/java/uk/gov/ons/census/caseapisvc/endpoint/CaseEndpointIT.java [hunk 3]
```diff
@@ -589,6 +589,7 @@ public void testGetNewIndividualUacQidForEnglishHouseholdCase()
   @DirtiesContext
   public void testGetNewUacQidForCaseDistributesFulfilmentEvent()
       throws UnirestException, IOException, InterruptedException {
+    // TODO This test can fail intermittently, needs improving
```
Comment (NickGrantX) [inline src/test/java/uk/gov/ons/census/caseapisvc/endpoint/CaseEndpointIT.java]: I can pair up with you, implementing the `QueueSpy` pattern in the CaseApi integration tests, which I think will massively improve things

File: src/main/java/uk/gov/ons/census/caseapisvc/model/dto/QidLink.java (hunk 1 above)
Comment (ryangrundy7) [reply src/main/java/uk/gov/ons/census/caseapisvc/model/dto/QidLink.java:9]: Added transaction id 👍

File: src/main/java/uk/gov/ons/census/caseapisvc/service/UacQidService.java (hunk 2 above)
Comment (ryangrundy7) [reply src/main/java/uk/gov/ons/census/caseapisvc/service/UacQidService.java:138]: Passed in ROPS_UI as channel

**Review by NickGrantX — APPROVED** [review]

**Review by NickGrantX — CHANGES_REQUESTED** [review]
ITs broken.

Need to add `case.rm.unInvalidateAddress` queue to rabbit config

**Review by NickGrantX — APPROVED** [review]
False alarm! All good

**Review by liamtoozer — APPROVED** [review]

**Pull request merged**